app/code/Magento/Customer/Model/AccountManagement.php

```diff
@@ -928,7 +928,19 @@ public function createAccountWithPasswordHash(CustomerInterface $customer, $hash
             }
             $this->customerRegistry->remove($customer->getId());
         } catch (InputException $e) {
+            $isSecureArea = $this->registry->registry('isSecureArea');
+            if ($isSecureArea !== null) {
+                $this->registry->unregister('isSecureArea');
+            }
+            $this->registry->register('isSecureArea', true);
+
```

Comment:
We should not simulate a secure area here, instead, we should decouple the customer creation with address creation. I mean, the customer should not be deleted in case there is a problem to save the address, since the customer object does not depend on the address object. On this case, we likely should proceed as usual in case of InputException and just maybe add an error message to user. What do you think?

Reply:
@gabrieldagama Hi, I can see your point of view, maybe I was looking at this through project-tinted glasses (in this particular store a customer always has to have an address, which is populated from a CRM).

In my opinion this PR could be closed. It is still confusing that the customer will be redirected back to registration and they cannot register with the same email due to the customer having been created already, but I don't think that's necessary an issue that needs to be fixed to M2 core. The customer can anyways log in to the store at that point already, they just have to navigate away from the "broken" registration page first.